tests/plenary/ui/mappings_spec.lua

```diff
@@ -1568,4 +1568,28 @@ describe('Mappings', function()
     vim.cmd('norm ,o,a\r')
     assert.are.same('* [#A] Test orgmode', vim.fn.getline(1))
   end)
+
+  it('should refile to headline that matches name exactly', function()
+    local destination_file = helpers.load_file_content({
+      '* foobar',
+      '* baz',
+      '* foo',
+    })
+
+    local source_file = helpers.load_file_content({
+      '* to be refiled',
+      '* not to be refiled',
+    })
+
+    vim.fn.cursor(1, 1)
+    vim.cmd([[norm ,or]])
+    -- TODO: this is wrong, not a command. how to do the actual refiling?
+    vim.cmd([[exe "destination.org/foo\<CR>"]])
+    assert.are.same('* not to be refiled', vim.fn.getline(1))
+    -- TODO: how to switch to the destination file?
```

Comment:
you could use a fixture for the destination file and load it using `vim.cmd[[ edit <destfile> ]]` then from there use `vim.api.nvim_buf_get_lines` to get the list of lines to assert against